### PR TITLE
Fix Gorilla Tactics

### PR DESCRIPTION
**Pull request by RoccoDev**
The issue was in the second branch of control flow:
```js
['Gmax', 'Dynamax'].indexOf(defender.name) !== -1
```
which doesn't make sense since Gorilla Tactics doesn't work only if the **attacker** is **not** dynamaxed.

### CODE REVIEW

File: calc/src/mechanics/gen8.ts [hunk 1]
```diff
@@ -791,7 +788,7 @@ function calculateGen8(
     description.attackerAbility = attacker.ability;
   } else if (
     attacker.hasAbility('Gorilla Tactics') &&
-    ['Gmax', 'Dynamax'].indexOf(defender.name) !== -1
+    ['Gmax', 'Dynamax'].indexOf(attacker.name) === -1
```
Comment (Zarel) [inline calc/src/mechanics/gen8.ts]: Is the attacker name literally going to be "Gmax" or "Dynamax"? I think this clearly needs two passes, so they can be searched.

Comment (RoccoDev) [reply calc/src/mechanics/gen8.ts]: I agree that we shouldn't use names to track maxing and should instead use a field to track it. What's the state of the Dynamax branch?

Comment (Zarel) [reply calc/src/mechanics/gen8.ts]: I have no clue, sorry, in this repo I'm just a rando who codes.

File: calc/src/mechanics/gen8.ts [hunk 2]
```diff
@@ -1053,6 +1050,10 @@ function calculateGen8(
     finalMods.push(0x800);
     description.defenderAbility = defender.ability;
   }
+  if (move.isSound && attacker.hasAbility('Punk Rock')) {
+    finalMods.push(0x1300);
```
Comment (AustinXII) [inline calc/src/mechanics/gen8.ts]: What’s wrong with 0x14cd?

Comment (AustinXII) [reply calc/src/mechanics/gen8.ts]: Also what do you mean by “checked twice”?

Comment (RoccoDev) [reply calc/src/mechanics/gen8.ts]: You're right, my bad, that was probably some leftover from testing. Changed it back to 0x14cd.

**Comment by AustinXII** [comment]
The only thing left is finding out what you meant by the  gorillla tactics check being checked twice

**Comment by AustinXII** [comment]
(I’m going off of one of your commits name btw)

**Comment by RoccoDev** [comment]
Yeah so, when I merged, I accidentally left another check for Punk Rock (the one with 0x1300) in the code. So when it was implemented elsewhere, both checks were present (in my local repository).
That commit in tandem with the latest one (that fixes 0x14cd) simply do nothing.

**Comment by AustinXII** [comment]
Oooh makes sense. Thanks!

File: calc/src/mechanics/gen8.ts (hunk 1 above)
Comment (AustinXII) [reply calc/src/mechanics/gen8.ts]: I can add a field to dynamax pokemon and update this once that branch is done

**Comment by RoccoDev** [comment]
I made something similar in #372

File: calc/src/mechanics/gen8.ts (hunk 1 above)
Comment (RoccoDev) [reply calc/src/mechanics/gen8.ts]: I made something similar in #372

File: calc/src/mechanics/gen8.ts (hunk 2 above)
Comment (AustinXII) [reply calc/src/mechanics/gen8.ts]: Sweet, just check the field for this check then :)

**Pull request merged**